buildtest/utils/file.py

```diff
@@ -143,3 +143,80 @@ def resolve_path(path):
 
     if os.path.exists(real_path):
         return real_path
+
+def read_file(filepath):
+    """ This method provides capability to read a file.
+
+        Parameters:
+
+        :param filepath: file name to read
+        :type filepath: str, required
+        :return: return content of file
+        :rtype: list
+    """
+
+    # type check filepath to ensure its a string, if not return None
+    if not isinstance(filepath,str):
+        return None
+
+    # resolve_path will handle shell and user expansion and account for any symlinks and check for file existence.
+    # if resolve_path does not return gracefully it implies file does not exist and will return None
+    filepath = resolve_path(filepath)
+
+    # if it's invalid file let's return None
+    if not filepath:
+        return None
+
+    content = []
+    fd = open(filepath, "r")
+
+    while True:
+
+        line = fd.readline()
+        if line:
+            content.append(line)
+        else:
+            break
+
+    return content
+
+def write_file(filepath, content):
+    """ This method provides capability to write a file.
+
+        Parameters:
+
+        :param filepath: file name to write
+        :type filepath: str, required
+        :param content: content to write to file
+        :type content: list, required
+        :return: return content of file
+        :rtype: list
+    """
+
+    # type check filepath to ensure its a string, if not return None
+    if not isinstance(filepath,str):
+        return None
+
+    # if content was passed as a string, let's convert to list
+    if isinstance(content, str):
+        content = content.splitlines(True)
+
+    # shown below is equivalent to running 'resolve_path' but we can't invoke it since we expect this method
+    # to write to a new file. resolve_path assumes filepath already exists and returns realpath otherwise returns
+    # None.
+    filepath = os.path.expanduser(filepath)
+    filepath = os.path.expandvars(filepath)
+    filepath = os.path.realpath(filepath)
```

Comment:
This is also not scoped to the function - we simply need to check if it's a string, and if the directory name to write it to exists, and then write it :) These functions should be 3-6 lines at most!